packages/components/phone-number/src/components/osds-phone-number/core/controller.ts

```diff
@@ -0,0 +1,56 @@
+import { ODS_COUNTRY_ISO_CODE, ODS_COUNTRY_ISO_CODES } from '@ovhcloud/ods-common-core';
+import { OsdsPhoneNumber } from '../osds-phone-number';
+
+class OdsPhoneNumberController {
+
+  constructor(private readonly component: OsdsPhoneNumber) { }
+
+  setDefaultLocale(): ODS_COUNTRY_ISO_CODE {
+    if (this.isOdsCountryCode(this.component.locale)) {
+      return ODS_COUNTRY_ISO_CODES[ODS_COUNTRY_ISO_CODES.indexOf(this.component.locale)];
+    }
+    const navigatorLang = this.getNavigatorLang();
+    if (navigatorLang) {
+        return ODS_COUNTRY_ISO_CODES[ODS_COUNTRY_ISO_CODES.indexOf(navigatorLang)];
+    }
+    return ODS_COUNTRY_ISO_CODE.FR;
+  }
+
+  setDefaultIsoCode(): ODS_COUNTRY_ISO_CODE {
+    if (this.isOdsCountryCode(this.component.isoCode)) {
+        return ODS_COUNTRY_ISO_CODES[ODS_COUNTRY_ISO_CODES.indexOf(this.component.isoCode)];
+    }
+    const navigatorLang = this.getNavigatorLang();
+    if (navigatorLang) {
+        return ODS_COUNTRY_ISO_CODES[ODS_COUNTRY_ISO_CODES.indexOf(navigatorLang)];
+    }
+    return ODS_COUNTRY_ISO_CODE.AD;
+  }
+
+  getCountriesList(): ODS_COUNTRY_ISO_CODE[] {
+    if (this.component.countries === 'all') {
+      return ODS_COUNTRY_ISO_CODES as ODS_COUNTRY_ISO_CODE[];
+    }
+    return this.component.countries || [];
+  }
+
+  private getNavigatorLang(): ODS_COUNTRY_ISO_CODE | undefined {
+    const language = navigator.language?.split('-')[0];
+    if (language && this.isOdsCountryCode(language)) {
+      return language;
+    }
+    const language2 = navigator.language?.split('-')[1];
```

Comment:
Why ? Because navigator.language is a string like `en-us`, `fr-fr` but `fr` it's an isoCode 👍 but `en` no 
And we have many language like `fr-be`, `fr-ca`, `en-uk`, `en-us`...

Reply:
Only first part of the the code should be used, the second part is to specify dialect but it's not meant to be an actual locale.
If the first part is not managed in our list, just directly fallback to default locale

Reply:
Yes but all the english country will not work correctly

Reply:
Issue is there is a difference between `isoCode` that we use for flag => indeed we need the second part
And the `locale` which is always `fr`, `en`.
But this is used on both function